### PR TITLE
#1609: [Refactoring] DateTime handling and formatting

### LINKED ISSUE #1609: [Refactoring] DateTime handling and formatting
Related to https://github.com/magento/adobe-stock-integration/issues/1578

As per code review proper DateTime handling and formatting should be used. Magento framework DateTime functionality should be utilized

Files to check:
- `MediaGalleryUi/Model/AssetDetailsProvider/CreatedAt.php` - Should it be configurable or inherited from the current Magento locale?
- `MediaGalleryUi/Model/AssetDetailsProvider/UpdatedAt.php`
- `MediaGalleryUi/Model/AssetDetailsProvider/CreatedAt.php`
- `MediaGallerySynchronization/Model/SynchronizeFiles.php`

### PR DESCRIPTION
**Pull request by yolouiese**
### Need to verify (*)
- **Created** and **Modified** values are saved and displayed without any issues when viewing image details

### Description (*)
- Refactored datetime handling and formatting by adding/using Magento Framework DateTime functionality 

### Fixed Issues (if relevant)
1. Fixes magento/adobe-stock-integration#1609: [Refactoring] DateTime handling and formatting

### Manual testing scenarios (*)
1. Go to **Content - Media Gallery**
2. Search for any image from the **Search Adobe Stock** button
2. Select the added image from Media Gallery and click "three dots" in the bottom right
3. Select **View Details**
4. Check **Created** and **Modified** values under **Details**

### CODE REVIEW

**Comment by yolouiese** [comment]
@magento run all tests

File: MediaGalleryUi/Model/AssetDetailsProvider/CreatedAt.php [hunk 1]
```diff
@@ -22,6 +21,20 @@ class CreatedAt implements AssetDetailsProviderInterface
...
+    /**
+     * @var DateTime
+     */
+    private $date;
+
+    /**
+     * @param DateTime $date
+     */
+    public function __construct(
+        DateTime $date
```
Comment (sivaschenko) [inline MediaGalleryUi/Model/AssetDetailsProvider/CreatedAt.php]: I think it's better to use `\Magento\Framework\Stdlib\DateTime\TimezoneInterface` for retrieving and formatting a date (only for view layer, not applicable to `SynchronizeFiles` class)

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the pull request @yolouiese ! Please take a look at my comment

**Comment by yolouiese** [comment]
@magento run all tests

**Review by sivaschenko — DISMISSED** [review]

**Comment by chalov-anton** [comment]
:heavy_check_mark: QA Passed

_Created_ and _Modified_ dates are displayed under Image Details
The _Modified_ date and time is updated after Editing

**Comment by sivaschenko** [comment]
@magento run all tests

**Pull request merged**